src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1292,14 +1297,30 @@ void MainWindow::slot_open_recent()
     }
 }
 
-void MainWindow::slot_clear_open_recent_files_menu()
+void MainWindow::slot_clear_recent_project_files_menu()
 {
     QSettings settings(SETTINGS_ORGANIZATION, SETTINGS_APPLICATION);
     settings.setValue("recent_file_list", QStringList());
 
     update_recent_files_menu(QStringList());
 }
 
+void MainWindow::slot_clear_recent_nonexistent_project_files()
+{
+    QSettings settings(SETTINGS_ORGANIZATION, SETTINGS_APPLICATION);
+    QStringList files = settings.value("recent_file_list").toStringList();
+    QStringList existent_files;
+
+    for (int i = 0; i < files.size(); i++)
+    {
+        if (QFileInfo(files[i]).isFile())
```

Comment:
I made sure `isFile` is appropriate for what we want: it return false for non-existent files and for directories.